### PR TITLE
Feature/integrate typeorm and add user table

### PR DESCRIPTION
**Pull request by Saela**
- Configure api to use typeorm
- Configure migrations
- Update docker file to run migrations before starting the application
- Update api to fetch the users from the database

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#########################################################
+#    _______                ____  _____  __  __
+#   |__   __|              / __ \|  __ \|  \/  |
+#      | |_   _ _ __   ___| |  | | |__) | \  / |
+#      | | | | | '_ \ / _ \ |  | |  _  /| |\/| |
+#      | | |_| | |_) |  __/ |__| | | \ \| |  | |
+#      |_|\__, | .__/ \___|\____/|_|  \_\_|  |_|
+#          __/ | |
+#         |___/|_|
+#########################################################
```
Comment (Tugark) [inline .env.example:10]: I mean it is cool, but seems a bit excessive 😆

File: .github/workflows/deployment.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ jobs:
   deploy:
     name: Build and deploy project
     runs-on: ubuntu-latest
+    environment: Staging
```
Comment (Tugark) [inline .github/workflows/deployment.yml:11]: What do we need this for?

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
-import { UsersService } from '../users/users.service';
 import { JwtService } from '@nestjs/jwt';
+import { UserService } from '../user/user.service';
 
 @Injectable()
 export class AuthService {
   constructor(
-    private usersService: UsersService,
+    private userService: UserService,
     private jwtService: JwtService,
```
Comment (Tugark) [inline src/auth/auth.service.ts:9]: I think we wanted to make these `readOnly`?

File: src/user/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { User } from './entities/user.entity';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @InjectRepository(User)
+    private userRepository: Repository<User>,
```
Comment (Tugark) [inline src/user/user.service.ts:10]: `readOnly`

**Review by Tugark — COMMENTED** [review]
LGTM

File: .env.example (hunk 1 above)
Comment (Saela) [reply .env.example:10]: LGTM ¯\_(ツ)_/¯

File: .github/workflows/deployment.yml (hunk 2 above)
Comment (Saela) [reply .github/workflows/deployment.yml:11]: We don't anymore. It was for setting the secrets from github. I'll remove it.

File: src/auth/auth.service.ts (hunk 3 above)
Comment (Saela) [reply src/auth/auth.service.ts:9]: Yes, forgot to change it.

File: src/user/user.service.ts (hunk 4 above)
Comment (Saela) [reply src/user/user.service.ts:10]: Not sure if this can be made read only. The NestJs documentation tells you to add it like this. I'll re-check it.

Comment (Saela) [reply src/user/user.service.ts:10]: Seems to be fine. Changed it.

File: src/auth/auth.service.ts (hunk 3 above)
Comment (Saela) [reply src/auth/auth.service.ts:9]: Cleaned up: https://github.com/gipfeli-io/gipfeli-api/pull/16

File: .github/workflows/deployment.yml (hunk 2 above)
Comment (Saela) [reply .github/workflows/deployment.yml:11]: Cleaned up: https://github.com/gipfeli-io/gipfeli-api/pull/16

**Pull request merged**